### PR TITLE
Support fp32 gating logits in topk_sigmoid

### PR DESCRIPTION
**Pull request by jmunetong**
Widen the topk_sigmoid dispatch from AT_DISPATCH_REDUCED_FLOATING_TYPES (Half, BFloat16 only) to AT_DISPATCH_FLOATING_TYPES_AND2 so float32 router gating logits (e.g. Nemotron-3-Nano MoE) are accepted. The kernel already upcasts each element to float internally, so the fp32 path is numerically a no-op cast.

Add test_topk_sigmoid_fp32 exercising the new fp32 path against the float reference across the same token/expert/topk/renormalize/bias grid.

### CODE REVIEW

File: tests/test_topk_sigmoid.py [hunk 1]
```diff
@@ -98,5 +98,62 @@ def test_topk_sigmoid(
     torch.testing.assert_close(res, ref, atol=atol, rtol=rtol)
 
 
+@pytest.mark.parametrize("n_token", [2, 32, 4096])
```
Comment (gaopengff) [inline tests/test_topk_sigmoid.py]: Better to reuse method 'test_topk_sigmoid' for fp32 test. You can add datatype with the following code:
"@pytest.mark.parametrize("dtype", [torch.bfloat16, torch.float16, **torch.float32**])"

Comment (jmunetong) [reply tests/test_topk_sigmoid.py]: Thank you! I have modified the test to include your feedback.

**Review by airMeng — APPROVED** [review]
